include/irr/asset/filters/CSummedAreaTableImageFilter.h

```diff
@@ -0,0 +1,402 @@
+			else
+				return executeInterprated(state, reinterpret_cast<double*>(state->scratchMemory));
+		}	
+
+	private:
+
+		template<typename decodeType> //!< double or uint64_t
+		static inline bool executeInterprated(state_type* state, decodeType* scratchMemory)
+		{
+			const asset::E_FORMAT inFormat = state->inImage->getCreationParameters().format;
+			const asset::E_FORMAT outFormat = state->outImage->getCreationParameters().format;
+			const auto inTexelByteSize = asset::getTexelOrBlockBytesize(inFormat);
+			const auto outTexelByteSize = asset::getTexelOrBlockBytesize(outFormat);
+			const auto currentChannelCount = asset::getFormatChannelCount(inFormat);
+			const auto arrayLayers = state->inImage->getCreationParameters().arrayLayers;
+			static constexpr auto maxChannels = 4u;
+
+			#ifdef _IRR_DEBUG
+			memset(scratchMemory, 0, state->scratchMemoryByteSize);
+			#endif // _IRR_DEBUG
+
+			const core::vector3du32_SIMD scratchByteStrides = [&]()
+			{
+				const auto trueExtent = vector3du32_SIMD(state->extent.width, state->extent.height, state->extent.depth);
+
+				switch (currentChannelCount)
+				{
+					case 1:
+					{
+						return TexelBlockInfo(asset::E_FORMAT::EF_R64_SFLOAT).convert3DTexelStridesTo1DByteStrides(trueExtent);
+					}
+
+					case 2:
+					{
+						return TexelBlockInfo(asset::E_FORMAT::EF_R64G64_SFLOAT).convert3DTexelStridesTo1DByteStrides(trueExtent);
+					}
+
+					case 3:
+					{
+						return TexelBlockInfo(asset::E_FORMAT::EF_R64G64B64_SFLOAT).convert3DTexelStridesTo1DByteStrides(trueExtent);
+					}
+					case 4:
+					{
+						return TexelBlockInfo(asset::E_FORMAT::EF_R64G64B64A64_SFLOAT).convert3DTexelStridesTo1DByteStrides(trueExtent);
+					}
+				}
+			}();
+			const auto scratchTexelByteSize = scratchByteStrides[0];
+
+			const auto&& [copyInBaseLayer, copyOutBaseLayer, copyLayerCount] = std::make_tuple(state->inBaseLayer, state->outBaseLayer, state->layerCount);
+			state->layerCount = 1u;
+
+			auto resetState = [&]()
+			{
+				state->inBaseLayer = copyInBaseLayer;
+				state->outBaseLayer = copyOutBaseLayer;
+				state->layerCount = copyLayerCount;
+			};
+
+			for (uint16_t w = 0u; w < copyLayerCount - copyInBaseLayer; ++w) 
+			{
+				std::array<decodeType, maxChannels> minDecodeValues = {};
+				std::array<decodeType, maxChannels> maxDecodeValues = {};
+
+				{
+					const auto inData = reinterpret_cast<const uint8_t*>(state->inImage->getBuffer()->getPointer());
+					const auto blockDims = asset::getBlockDimensions(state->inImage->getCreationParameters().format);
+					
+					auto decode = [&](uint32_t readBlockArrayOffset, core::vectorSIMDu32 readBlockPos) -> void
+					{
+						auto newReadBlockPos = (readBlockPos - decltype(readBlockPos)(state->inOffset.x, state->inOffset.y, state->inOffset.z)); // not sure
+
+						auto localOutPos = newReadBlockPos * blockDims;
+						auto* inDataAdress = inData + readBlockArrayOffset;
```

Comment:
TODO: follow https://github.com/buildaworldnet/IrrlichtBAW/pull/497#discussion_r470187251

Reply:
what you have right no is correct for reading "from image"